acquisition.py

```diff
@@ -2,113 +2,128 @@
         else:
             raise NotImplementedError
-    
+
     def get_reward(self, inputs_af_base):
         outputs = self.acq.get_reward_batch(inputs_af_base)
-        return  outputs
-    
+        return outputs
 
-'''
+
+"""
 BASE CLASS FOR ACQUISITION
-'''
+"""
+
+
 class AcquisitionFunctionBase:
-    '''
+    """
     Cf Oracle class : generic AF class which calls the right AF sub_class
-    '''
+    """
+
     @abstractmethod
     def __init__(self, config, proxy):
         self.config = config
         self.proxy = proxy
         self.device = self.config.device
-        
+
+        # the specific class of the exact AF is instantiated here
+
     @abstractmethod
     def load_best_proxy(self):
-        '''
+        """
         In case, loads the latest version of the proxy (no need normally)
-        '''
+        """
         if os.path.exists(self.config.path.model_proxy):
             self.proxy.load_model(self.config.path.model_proxy)
-        
+
         else:
             raise FileNotFoundError
 
     @abstractmethod
     def get_reward_batch(self, inputs_af_base):
-        '''
+        """
         calls the get_reward method of the appropriate Acquisition Class (MUtual Information, Expected Improvement, ...)
-        '''
-        raise NotImplementedError
-    
-    @abstractmethod
-    def base2af(self, inputs_af_base):
-        raise NotImplementedError
+        """
+        pass
 
 
-'''
+"""
 SUBCLASS SPECIFIC ACQUISITION
-'''
+"""
+
 
 class AcquisitionFunctionProxy(AcquisitionFunctionBase):
     def __init__(self, config, proxy):
         super().__init__(config, proxy)
-    
+
     def load_best_proxy(self):
         super().load_best_proxy()
 
-    
-    def get_reward_batch(self, inputs_af_base): #inputs_af = list of ...
-        inputs_af = list(map(self.base2af, inputs_af_base))
-        inputs = torch.stack(inputs_af).view(len(inputs_af_base), -1)
+    def get_reward_batch(self, inputs_af_base):  # inputs_af = list of ...
+        super().get_reward_batch(inputs_af_base)
+
+        inputs_af = list(map(torch.tensor, inputs_af_base))
+        input_afs_lens = list(map(len, inputs_af_base))
+        inputs = pad_sequence(inputs_af, batch_first=True, padding_value=0.0)
+        # inputs = self.proxy.base2proxy(inputs_af)
+
+        # inputs = torch.stack(inputs_af).view(len(inputs_af_base), -1)
+
         self.load_best_proxy()
         self.proxy.model.eval()
         with torch.no_grad():
-            outputs = self.proxy.model(inputs)
+            if self.config.proxy.model.lower() == "mlp":
+                outputs = self.proxy.model(inputs)
+            elif self.config.proxy.model.lower() == "lstm":
+                outputs = self.proxy.model(inputs, input_afs_lens)
+            elif self.config.proxy.model.lower() == "transformer":
+                outputs = self.proxy.model(inputs, None)
+
         return outputs
 
     def base2af(self, state):
```

Comment:
I agree that base2af could be removed because the input format for the acquisition function is going to be the input format of the proxy. So instead of implementing a base2af for each acquisition function, we can just take "preprocess" from each proxy !

Reply:
Yeah I agree. So I should delete `base2af()`? Ah, I don't think I understand the second part of your comment. Do you suggest explicitly calling the `preprocess()` function? In the current implementation, I am calling `forward()` and `preprocess()` is called implicitly inside `forward()`